### PR TITLE
.cursorrules V1

### PR DESCRIPTION
**Pull request by 0xDEnYO**
Just trying to kick Cursor's butt...:)

### CODE REVIEW

File: conventions.backup.md
Comment (0xDEnYO) [inline conventions.backup.md:1]: this is the old (very large) version of our conventions file

File: conventions.backup.md [hunk 1]
```diff
@@ -0,0 +1,849 @@
...
+
+  - All contracts in `src/` directory (except external dependencies)
+  - All test files in `test/solidity/` directory
+  - All deployment and task scripts in `script/` directory
+
+  License identifier format:
+
+  ```solidity
+  // SPDX-License-Identifier: LGPL-3.0-only
+  ```
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply conventions.backup.md]: the backup file is the old version and thus not so relevant anymore, we just keep it for now while testing the new version

File: .cursor/rules/05-solidity-basics.mdc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+name: Solidity baseline
+description: Baseline rules for all Solidity files (contracts, scripts, tests)
+globs:
+  - '**/*.sol'
+---
+
+- First line SPDX `LGPL-3.0-only` for our code (keep originals for third-party copies) immediately followed by `pragma solidity ^0.8.17;`.
+- Follow [CONV:BLANKLINES] and [CONV:NAMING]; keep solhint config in mind.
```
Comment (mirooon) [inline .cursor/rules/05-solidity-basics.mdc]: for solhint file here we can show cursor exact path to file (`.solhint.json`)

File: .cursor/rules/11-facets.mdc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - 'src/Facets/**/*.sol'
+---
+
+@15-architecture.mdc
+@16-security.mdc
+@17-gas.mdc
+
+- Required functions: `_startBridge` (internal), `swapAndStartBridgeTokensVia{FacetName}`, `startBridgeTokensVia{FacetName}`; attach `nonReentrant`, `refundExcessNative`, `validateBridgeData`, and appropriate does/don'tContain swap/call modifiers.
+- Events: emit `LiFiTransferStarted` inside `_startBridge` after validations/external calls; never emit `LiFiTransferCompleted` or `LiFiTransferRecovered` in facets; use `GenericSwapCompleted` for same-chain swaps when relevant.
+- Parameter rules: `{facetName}Data.receiverAddress` first and validated against `bridgeData.receiver` (EVM); check `targetChainId` vs `bridgeData.destinationChain`; non-EVM uses `bytes` receiver, non-zero, and `bridgeData.receiver == NON_EVM_ADDRESS`.
```
Comment (mirooon) [inline .cursor/rules/11-facets.mdc]: for bridging to a nonEVM chain, I would also include guidelines regarding the requirement to emit the BridgeToNonEVMChainBytes32 event

Comment (mirooon) [reply .cursor/rules/11-facets.mdc]: also check in case of nonEVM check _glacisData.receiverAddress == bytes32(0) with revert InvalidNonEVMReceiver

File: .cursor/rules/12-solidity-tests.mdc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+name: Solidity tests
+description: Foundry test structure, naming, and expectations
+globs:
+  - "test/**/*.t.sol"
+---
+- Require `setUp()`; mirror `src/` layout; call `initTestBase()` when inheriting `TestBase`; label actors with `vm.label`.
```
Comment (mirooon) [inline .cursor/rules/12-solidity-tests.mdc]: for facets, I would add that we should inherit from TestBaseFacet rather than TestBase, since standard facet functions need to be overridden 

and also if a facet does not support native tokens, it may leave tests such as testBase_CanSwapAndBridgeNativeTokens and testBase_CanBridgeNativeTokens empty, provided it includes a brief explanation clarifying why these tests are intentionally left unimplemented.

File: .cursor/rules/11-facets.mdc [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+description: Facet-only requirements and validations
+globs:
+  - 'src/Facets/**/*.sol'
+---
+
+@15-architecture.mdc
+@16-security.mdc
+@17-gas.mdc
+
+- Required functions: `_startBridge` (internal), `swapAndStartBridgeTokensVia{FacetName}`, `startBridgeTokensVia{FacetName}`; attach `nonReentrant`, `refundExcessNative`, `validateBridgeData`, and appropriate does/don'tContain swap/call modifiers.
```
Comment (mirooon) [inline .cursor/rules/11-facets.mdc]: I would also add a rule abou constructor arguments that zero values should not be allowed. Code should validate addresses against the zero address and numeric parameters against zero

File: .cursor/rules/11-facets.mdc [hunk 6]
```diff
@@ -0,0 +1,16 @@
+---
+name: Diamond facets
```
Comment (mirooon) [inline .cursor/rules/11-facets.mdc]: do you know how Cursor selects these rules files? Is it based on the filename? For example if I type in the chat ‘implement a new diamond contract…’, will it correctly detect that it’s a facet and apply this rules file? We may want to be more explicit and name it something like ‘Diamond facet contracts’ to ensure proper matching. wdyt?

File: .cursor/rules/13-solidity-interfaces.mdc [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---
+name: Solidity interfaces
+description: Interface-only rules for Solidity interfaces in src/Interfaces
+globs:
+  - 'src/Interfaces/**/*.sol'
+---
+
+- Production interfaces (used in `src/`) must be Solidity `.sol` files located in `src/Interfaces/`; do not place production interfaces elsewhere. Test files may define inline interfaces as needed for testing purposes. Do not mix interface and implementation in the same file.
```
Comment (mirooon) [inline .cursor/rules/13-solidity-interfaces.mdc]: "Do not mix interface and implementation in the same file." - I would add that it is allowed for test files

File: .cursor/rules/20-solidity-scripts.mdc [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+name: Solidity deploy/update scripts
+description: Foundry deployment/update script patterns
+globs:
+  - 'script/**/*.s.sol'
+  - 'script/**/*.sol'
+---
+
+- Use `DeployScriptBase`/`UpdateScriptBase` patterns: `Deploy*.s.sol` uses `deploy(type(Contract).creationCode)`; only add `getConstructorArgs()` when needed and read from JSON via `stdJson`.
+- Update scripts call `update(\"{ContractName}\")` and override `getExcludes()` for selector skips.
```
Comment (mirooon) [inline .cursor/rules/20-solidity-scripts.mdc]: what about giving more context here for getExcludes() and add if selectors from other facet's can conflicts when calling diamond for example or functions that we want to call only on diamondCut?

File: .cursor/rules/20-solidity-scripts.mdc [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+- Keep naming/location under `script/deploy/` (and `zksync/` when relevant); reuse helpers/config files (`deployRequirements.json`, `targetState.json`) instead of new patterns.
```
Comment (mirooon) [inline .cursor/rules/20-solidity-scripts.mdc]: since we often have duplicated scripts for zksync, I think it would be helpful if cursor, when applying changes, also checked the zksync/ folder to ensure that any modification made in the main non-zksync script is reflected in the zksync script. This is a common case. Of course there are cases where we don’t want the zksync version to change, but I believe it’s better to review and discard an unnecessary change than to accidentally overlook the zksync script entirely (it happened for me many times)
In most situations both versions follow the same pattern, although there are some differences in some cases

File: .cursor/rules/30-typescript.mdc [hunk 10]
```diff
@@ -0,0 +1,14 @@
+---
+name: TypeScript scripts
+description: TS scripting, helpers, lint/tests
+globs:
+  - 'script/**/*.ts'
+  - 'tasks/**/*.ts'
+---
+
+- Follow [CONV:SCRIPTS-TS] for TS script conventions; run with `bunx tsx`; obey `.eslintrc.cjs`; avoid `any`; use TypeChain types (e.g., `ILiFi.BridgeDataStruct`).
+- Always reuse existing helper functions from `script/utils/`, `script/demoScripts/utils/`, and other helper directories; prefer existing patterns over new ones. Search for existing helpers before implementing new functionality.
```
Comment (mirooon) [inline .cursor/rules/30-typescript.mdc]: I would explicitly add `script/common/` as well

File: conventions.md [hunk 11]
```diff
@@ -1,849 +1,258 @@
...
+- **Add the enforceable bits** to `.cursor/rules/*.mdc` and/or `docs/conventions_digest/*` with new `[CONV:*]` anchors.
+- Keep this file focused on:
+  - Architectural shape (where things live, how they interact).
+  - Governance/security guarantees.
+  - High-level rationale and tradeoffs between options.
+
+If a new pattern becomes common:
+
+- Add a **short digest entry** (`docs/conventions_digest/*`) for agents.
+- Update `.cursor/rules` to reference that digest instead of copying long explanations.
```
Comment (mirooon) [inline conventions.md]: *I picked random place to write comment*

what about adding more context regarding mongodb and proposals? for example if we will want to update mongodb scripts with proposal related data, cursor will have additional context to work with

File: .cursor/rules/11-facets.mdc [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+globs:
+  - 'src/Facets/**/*.sol'
+---
+
+@15-architecture.mdc
+@16-security.mdc
+@17-gas.mdc
+
+- Required functions: `_startBridge` (internal), `swapAndStartBridgeTokensVia{FacetName}`, `startBridgeTokensVia{FacetName}`; attach `nonReentrant`, `refundExcessNative`, `validateBridgeData`, and appropriate does/don'tContain swap/call modifiers.
+- Events: emit `LiFiTransferStarted` inside `_startBridge` after validations/external calls; never emit `LiFiTransferCompleted` or `LiFiTransferRecovered` in facets; use `GenericSwapCompleted` for same-chain swaps when relevant.
```
Comment (mirooon) [inline .cursor/rules/11-facets.mdc]: and maybe not here but some explanation about: `script/deploy/resources/deployRequirements.json`

File: docs/conventions_digest/scripts_and_ci.md [hunk 13]
```diff
@@ -2,13 +2,13 @@
...
-  sendTransaction, ensureBalanceAndAllowanceToDiamond, getUniswapData*).
-
-- 100% unit test coverage for new TS helpers using Bun.
+- TS scripts use `.eslintrc.cjs` rules, `bunx tsx`, `citty`, `consola`, and env validated via helpers (e.g., `getEnvVar()`).
+- MUST use viem for all contract interactions in demo/operational scripts; ethers.js helpers are deprecated.
+- ALWAYS reuse existing helpers where available, in particular:
+  - `script/utils/deploymentHelpers.ts` (deployment loading helpers),
+  - `script/demoScripts/utils/demoScriptHelpers.ts` (viem-based demo helpers and swap helpers).
+- DO NOT use deprecated ethers-based helpers (`getProvider`, `getWalletFromPrivateKeyInDotEnv`, ethers `sendTransaction`, `ensureBalanceAndAllowanceToDiamond`).
+- New TypeScript helpers require colocated `{name}.test.ts` with 100% coverage using Bun (`describe` / `it` / `expect`).
```
Comment (mirooon) [inline docs/conventions_digest/scripts_and_ci.md]: in solidity_style.md under CONV:BLANKLINES I would add all previous rules for tests from old conventions.md:
     - Blank line between `vm.expectRevert()` and function call
     - Blank line before `vm.stopPrank()` if separate logical block
     - Blank line before assertions
     - Single blank line between test cases
     - Single blank line after `vm.startPrank(address)`
     - For `vm.expectEmit` blocks: no blank line between `vm.expectEmit` and its event definition, but blank lines before and after the entire block
     - Group related test assertions together without blank lines between them

Comment (mirooon) [reply docs/conventions_digest/scripts_and_ci.md]: also _Variable Naming Conventions_ section is missing and imo valuable

File: .cursor/rules/12-solidity-tests.mdc [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+- Names: `test_` (success), `testRevert_` (failures), `testBase_` (base). Use `vm.expectRevert` with specific reason and `vm.expectEmit(true, true, true, true, <addr>)` for events.
+- Structure: setup → execute → assert; apply [CONV:BLANKLINES] test rules (gap after `vm.expectRevert` before call, gap before assertions/events).
+- For whitelist flows inherit `TestWhitelistManagerBase` and use `addToWhitelist` / `setFunctionWhitelistBySelector` helpers.
+- After test changes, run `forge test` (or narrowed scope) or state which suites remain.
```
Comment (mirooon) [inline .cursor/rules/12-solidity-tests.mdc]: for tests files I suggest to add point for import ordering. system libraries first, then project files

File: .cursor/rules/30-typescript.mdc [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+- Do NOT use deprecated ethers-based helpers (`getProvider`, `getWalletFromPrivateKeyInDotEnv`, ethers `sendTransaction`, `ensureBalanceAndAllowanceToDiamond`); use viem-based flows instead.
+- New helpers require colocated `{name}.test.ts` with 100% coverage using Bun (`describe`/`it`/`expect`); cover edge cases, type safety, and error paths.
+- CLI: use `citty`; logging via `consola`; validate env via `getEnvVar()`; exit 0/1 appropriately.
+- After TS changes run lint/tests (or state which remain) to ensure eslint/type checks pass.
```
Comment (mirooon) [inline .cursor/rules/30-typescript.mdc]: probably it's work to mention that if we will prompt 'demo script' then we mean script that it's inside 'script/demoScripts' and that usually they follow similar patterns in terms of how they are structured. And we can give references to demoLidoWrapper.ts, demoUnit.ts, demoEco.ts

File: .cursor/rules/11-facets.mdc (hunk 12 above)
Comment (mirooon) [reply .cursor/rules/11-facets.mdc]: that it:
-Controls deployment rules and dependencies
-Manages contract dependencies
-Handles network-specific parameters
-Specifies zero address restrictions
-Defines required external config files
-Usage in helperFunctions.sh

File: .cursor/rules/README.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+- `12-*`: Test rules
+- `15-*`: Architecture rules
+- `16-*`: Security rules
+- `17-*`: Performance rules (gas)
+- `18-*`: Testing discipline
+- `20-*`: Script rules (Solidity)
+- `30-*`: TypeScript rules
+- `40-*`: Bash rules
+- `99-*`: Final checks (completion)
+
```
Comment (mirooon) [inline .cursor/rules/README.md:16]: what about rules about audit files and github workflows. We had whole sections before

File: script/playgroundHelpers.sh [hunk 17]
```diff
@@ -712,32 +712,34 @@ function compareContractBytecode() {
...
+  # Resolve premium RPC URL from environment (.env / CI secrets) via helper
+  local RPC_URL
+  RPC_URL=$(getRPCUrl "$NETWORK")
+  if [[ $? -ne 0 || -z "$RPC_URL" ]]; then
+    error "[$NETWORK] Failed to resolve RPC URL via getRPCUrl"
     return 1
   fi
 
-  echo "Analyzing transaction: $TX_HASH with RPC URL: $RPC_URL"
+  echo "Analyzing transaction: $TX_HASH on network: $NETWORK with RPC URL: $RPC_URL"
```
Comment (mirooon) [inline script/playgroundHelpers.sh:742]: good job 💯 ! Looking forward to use it on the battlefield along with analyzer. I am curious!

File: .cursor/rules/003-context-monitor.mdc
Comment (rjblopes) [inline .cursor/rules/003-context-monitor.mdc:1]: Not sure if this should be a rule by it self, at least applying all the time. 🤔

File: .cursor/rules/000-global-standards.mdc
Comment (rjblopes) [inline .cursor/rules/000-global-standards.mdc:1]: Looks good but I think it's missing file structure details, i.e. explain how to split and where to add files. Also we should include in globals stuff like: code quality rules, conventions, security concerns, architecture, etc...

File: .cursor/rules/11-facets.mdc (hunk 6 above)
Comment (rjblopes) [reply .cursor/rules/11-facets.mdc]: Cursor either selects it automatically, based in the `globs` and `alwaysApply: true`, or you have to reference the rule with `@11-facets.mdc`. I don't think naming it different will impact.

File: .cursor/rules/11-facets.mdc [hunk 18]
```diff
@@ -0,0 +1,16 @@
+---
+name: Diamond facets
+description: Facet-only requirements and validations
+globs:
+  - 'src/Facets/**/*.sol'
+---
+
+@15-architecture.mdc
+@16-security.mdc
+@17-gas.mdc
```
Comment (rjblopes) [inline .cursor/rules/11-facets.mdc]: Unless these rules have different globs, or we want to use manually on other files, we should be very cautious about importing other rules to save AI tokens. This means potential duplicate information will be injected in the context.

File: .cursor/rules/15-architecture.mdc [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+name: Architecture and design
+description: Diamond, periphery, helpers, and governance architecture guardrails
+globs:
+  - 'src/**/*.sol'
+  - 'script/**/*.sol'
+  - 'script/**/*.ts'
+  - 'test/**/*.t.sol'
+---
+
+- Follow the high-level architecture described in `conventions.md` under “Architectural Principles”, “Diamond Standard (EIP-2535)”, and “Facet Contract Requirements” instead of inventing new patterns.
```
Comment (rjblopes) [inline .cursor/rules/15-architecture.mdc]: All these references should be rules or links to files. For me conventions and architecture principles make sense to be in global rules (always apply).

File: .cursor/rules/099-finish.mdc
Comment (rjblopes) [inline .cursor/rules/099-finish.mdc:1]: Candidate to be moved to global section

File: .cursor/rules/transaction_analysis.cursorrules.mdc
Comment (rjblopes) [inline .cursor/rules/transaction_analysis.cursorrules.mdc:1]: Please use naming conventions for all rules files

File: conventions.md [hunk 20]
```diff
@@ -1,849 +1,258 @@
...
-- Changes require multiple signatures
-- Moving towards Timelock-based governance:
-  1. Propose change to multisig
-  2. Collect required signatures
-  3. Execute call to Timelock
-  4. After minDelay period, anyone can execute the change
-
-# Codebase Structure
-
-    /lifi
```
Comment (rjblopes) [inline conventions.md:48]: This is file tree is a good candidate to go in a global rule

File: .cursor/commands/analyze-tx.md
Comment (rjblopes) [inline .cursor/commands/analyze-tx.md:1]: Don't know the reason for this file, but maybe this can be converted into a slash command in Cursor.

**Review by rjblopes — COMMENTED** [review]
Looks good and I think you captured the idea of `.mdc` rules files. However I see room for improvements. 

The convention naming and structure can reflect the groups more precisely. I suggest increase to 3 digits numbers prefix and enforce grouping based on first digit, eg `0xx-` global rules, `1xx-`solidity specific rules, `2xx-` typescript specific rules, etc...

Global rules should include repository structure description, architecture details, conventions, code quality, decisions, etc...

Rules should include code samples when possible/suitable. Including examples of what not to do.

Revise rules-in-rules references to avoid duplicate context and save on AI tokens.

Also made some comments throughout.

File: .cursor/commands/analyze-tx.md
Comment (ezynda3) [reply .cursor/commands/analyze-tx.md:1]: I second this. Would make adoption much easier https://cursor.com/docs/agent/chat/commands

File: .cursor/rules/11-facets.mdc (hunk 5 above)
Comment (ezynda3) [reply .cursor/rules/11-facets.mdc]: I think there are some legitimate cases where 0 address should still be allowed

File: .cursor/rules/12-solidity-tests.mdc [hunk 21]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (ezynda3) [inline .cursor/rules/12-solidity-tests.mdc]: Might also add a note to check base tests first before generating new tests to avoid creating duplicate tests

File: .cursor/rules/20-solidity-scripts.mdc (hunk 8 above)
Comment (0xDEnYO) [reply .cursor/rules/20-solidity-scripts.mdc]: good point, updated

File: .cursor/rules/20-solidity-scripts.mdc (hunk 9 above)
Comment (0xDEnYO) [reply .cursor/rules/20-solidity-scripts.mdc]: good point, added

File: .cursor/rules/30-typescript.mdc (hunk 10 above)
Comment (0xDEnYO) [reply .cursor/rules/30-typescript.mdc]: added

File: .cursor/rules/30-typescript.mdc (hunk 15 above)
Comment (0xDEnYO) [reply .cursor/rules/30-typescript.mdc]: added

File: .cursor/rules/12-solidity-tests.mdc (hunk 21 above)
Comment (0xDEnYO) [reply .cursor/rules/12-solidity-tests.mdc]: good point, added

File: .cursor/commands/analyze-tx.md
Comment (0xDEnYO) [reply .cursor/commands/analyze-tx.md:1]: I added a slash command for it and moved all the logic in there. 
However, AI also recommended to keep the .mdc rule file as it facilitates natural language invocation (and references the slash command)

File: .cursor/rules/transaction_analysis.cursorrules.mdc
Comment (0xDEnYO) [reply .cursor/rules/transaction_analysis.cursorrules.mdc:1]: good point, updated

File: .cursor/rules/003-context-monitor.mdc
Comment (0xDEnYO) [reply .cursor/rules/003-context-monitor.mdc:1]: why not? What is your concern here @rjblopes ?

I am curious to test this out and see if it works and actually warns when context is filling up. 
also printing rule names will help to see when it did loose context or not apply rules but should have.

File: .cursor/rules/000-global-standards.mdc
Comment (0xDEnYO) [reply .cursor/rules/000-global-standards.mdc:1]: added 001-project-structure.mdc

File: .cursor/rules/11-facets.mdc (hunk 18 above)
Comment (0xDEnYO) [reply .cursor/rules/11-facets.mdc]: removed them

File: .cursor/rules/099-finish.mdc
Comment (0xDEnYO) [reply .cursor/rules/099-finish.mdc:1]: moved

File: conventions.md (hunk 20 above)
Comment (0xDEnYO) [reply conventions.md:48]: added (and extended)

File: .cursor/rules/README.md (hunk 16 above)
Comment (0xDEnYO) [reply .cursor/rules/README.md:16]: good point, added

File: .cursor/rules/11-facets.mdc (hunk 6 above)
Comment (0xDEnYO) [reply .cursor/rules/11-facets.mdc]: lets test it out and see how it works....we can still adapt in coming versions.

File: .cursor/rules/12-solidity-tests.mdc (hunk 4 above)
Comment (0xDEnYO) [reply .cursor/rules/12-solidity-tests.mdc]: added

File: .cursor/rules/11-facets.mdc (hunk 5 above)
Comment (0xDEnYO) [reply .cursor/rules/11-facets.mdc]: yes, I agree. The GenericSwapFacetV3 for example has a nativeAddress parameter that in fact is address(0) on most networks

File: .cursor/rules/11-facets.mdc (hunk 3 above)
Comment (0xDEnYO) [reply .cursor/rules/11-facets.mdc]: good points, both added

File: .cursor/rules/05-solidity-basics.mdc (hunk 2 above)
Comment (0xDEnYO) [reply .cursor/rules/05-solidity-basics.mdc]: good point, updated

File: .cursor/rules/12-solidity-tests.mdc (hunk 14 above)
Comment (0xDEnYO) [reply .cursor/rules/12-solidity-tests.mdc]: added

File: .cursor/rules/15-architecture.mdc (hunk 19 above)
Comment (0xDEnYO) [reply .cursor/rules/15-architecture.mdc]: updated

File: .cursor/rules/003-context-monitor.mdc
Comment (0xDEnYO) [reply .cursor/rules/003-context-monitor.mdc:1]: discussed with Ricardo...we will try out this approach and learn

File: docs/conventions_digest/scripts_and_ci.md (hunk 13 above)
Comment (0xDEnYO) [reply docs/conventions_digest/scripts_and_ci.md]: added

File: .cursor/rules/100-solidity-basics.mdc [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+- Function params use leading underscore (e.g., `_amount`).
+
+## NatSpec Requirements ([CONV:NATSPEC])
+
+- Contracts & interfaces must include:
+  - `@title` - must match the contract/interface name (same as filename without `.sol`)
+  - `@author LI.FI (https://li.fi)`
+  - `@notice` - must describe what the contract/interface contains and its purpose
+  - `@custom:version X.Y.Z`
+- Public/external functions require NatSpec including params/returns.
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .cursor/rules/100-solidity-basics.mdc:26]: please update your own learning, these are the (new) rules

File: agent.md
Comment (ezynda3) [inline agent.md:1]: I think the conventional name is actually `AGENTS.md`

Comment (0xDEnYO) [reply agent.md:1]: uff, good catch,. Thx

Comment (0xDEnYO) [reply agent.md:1]: updated

File: .cursor/rules/README.md [hunk 23]
```diff
@@ -0,0 +1,110 @@
...
+| `401-testing-patterns.mdc`     | Cross-language testing, coverage, and structure                 | ❌ On match | `test/**/*.t.sol`, `**/*.test.ts`, `script/**/*.ts`                                                           |
+| `402-typescript-tests.mdc`     | Bun test structure and expectations for `.test.ts`              | ❌ On match | `**/*.test.ts`                                                                                                |
+| `500-github-actions.mdc`       | GitHub Actions workflows structure and conventions              | ❌ On match | `.github/workflows/**/*.yml`, `.github/workflows/**/*.yaml`, `.github/**/*.yml`, `.github/**/*.yaml`          |
+| `501-audits.mdc`               | Audit log and audit report management                           | ❌ On match | `audit/**/*.json`, `audit/**/*.pdf`, `.github/workflows/**/*audit*.yml`, `.github/workflows/**/*version*.yml` |
+| `600-transaction-analysis.mdc` | Transaction analysis activation gate                            | ❌ On match | `**/*`                                                                                                        |
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .cursor/rules/README.md:110]: updated

**Review by ezynda3 — APPROVED** [review]
